packages/react-components/src/accordion/src/useAccordionItems.ts

```diff
@@ -1,40 +1,61 @@
-import { Children, useMemo } from "react";
+import { Children, JSXElementConstructor, ReactElement, ReactNode, Ref, RefAttributes, useMemo } from "react";
 import { Content, Header } from "../../placeholders";
 import { isNil } from "lodash";
 import { mergeProps } from "../../shared";
 
+export interface AccordionBuilderItem {
+    id: string;
+    key: string;
+    index: number;
+    header: AccordionBuilderHeaderProps;
+    panel: AccordionBuilderPanelProps;
+}
+
+export interface AccordionBuilderHeaderProps {
+    elementType: string | JSXElementConstructor<any>;
+    ref: Ref<any>;
+    props: Record<string, any>;
+}
+
+
+export interface AccordionBuilderPanelProps {
+    elementType: string | JSXElementConstructor<any>;
+    ref: Ref<any>;
+    props: any;
+}
+
 export class AccordionBuilder {
     _rootId;
 
-    constructor(rootId) {
+    constructor(rootId: string) {
         this._rootId = rootId;
     }
 
-    build(children) {
+    build(children: ReactNode): AccordionBuilderItem[] {
         if (isNil(children)) {
             throw new Error("An accordion must have children.");
         }
 
-        return Children.map(children, (element, index) => {
-            const [header, content] = Children.toArray(element.props.children);
+        return Children.map(children, (element: ReactElement, index) => {
+            const [header, content] = Children.toArray(element.props.children) as ReactElement[];
 
             if (isNil(header) || isNil(content)) {
                 throw new Error("An accordion item must have an <Header> and a <Content>.");
             }
 
-            const key = !isNil(element.key) ? element.key.replace(".", "").replace("$", "") : index.toString();
+            const key = !isNil(element.key) ? (element.key as string).replace(".", "").replace("$", "") : index.toString();
 
```

Comment:
element.key can be a string or a number. This might be something we want to check in the future